service/src/main/java/bio/terra/cbas/runsets/monitoring/SmartRunsPoller.java

```diff
@@ -122,7 +125,12 @@ private void updateDatabaseRunStatus(
       if (r.status() != updatedRunState) {
         if (updatedRunState == CbasRunStatus.COMPLETE) {
           try {
-            updateOutputAttributes(r);
+            List<WorkflowOutputDefinition> outputDefinitionList =
+                objectMapper.readValue(
+                    r.runSet().method().outputDefinition(), new TypeReference<>() {});
+            if (!outputDefinitionList.isEmpty()) {
+              updateOutputAttributes(r, outputDefinitionList);
+            }
```

Comment:
This presumably is here to avoid empty PATCHes if there were no output definitions? Maybe we could add a comment for that?